packages/apps/dashboard/ui-2024/src/features/Leaderboard/hooks/useDataGrid.tsx

```diff
@@ -0,0 +1,201 @@
+import { Box, Typography } from '@mui/material';
+import { GridColDef, GridRenderCellParams } from '@mui/x-data-grid';
+import { LeaderBoardData } from '@services/api/use-leaderboard-details';
+import { useBreakPoints } from '@utils/hooks/use-is-mobile';
+import { useLeaderboardSearch } from '@utils/hooks/use-leaderboard-search';
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
+import CustomTooltip from '@components/CustomTooltip';
+import { colorPalette } from '@assets/styles/color-palette';
+import { useMemo } from 'react';
+
+import { RoleCell } from '../components/RoleCell';
+import { AddressCell } from '../components/AddressCell';
+import { ChainCell } from '../components/ChainCell';
+import { SelectNetwork } from '../components/SelectNetwork';
+import { ReputationLabel } from '../components/ReputationLabel';
+
+export const useDataGrid = (data: LeaderBoardData) => {
+	const {
+		filterParams: { chainId },
+	} = useLeaderboardSearch();
+
+	const {
+		mobile: { isMobile },
+	} = useBreakPoints();
+	const formattedData = useMemo(() => {
+		return data.map((row, index) => {
+			return {
+				...row,
+				id: index,
+			};
+		});
+	}, [data]);
+
+	const visibleRows = useMemo(() => {
+		if (chainId !== -1) {
+			return formattedData.filter((elem) => elem.chainId === chainId);
+		}
+
+		return formattedData;
+	}, [formattedData, chainId]);
+
+	const columns: GridColDef<(typeof visibleRows)[number]>[] = [
```

Comment:
I'm ok with moving it into useMemo hook, good catch, I memoized visibleRows and forgot about it. I'm not sure about to move somewhere else, because this is hook with logic for this datagrid so I think filters, sorting functions and columns definitions should be together. What do you think?